src/logic/publish.utils.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  assertMapDocumentNode,
+  assertProfileDocumentNode,
+  MapDocumentNode,
+  ProfileDocumentNode,
+} from '@superfaceai/ast';
+import {
+  parseProviderJson,
+  Parser,
+  ProviderJson,
+  SuperJson,
+} from '@superfaceai/one-sdk';
+import { getProfileOutput, validateMap } from '@superfaceai/parser';
+
+import { userError } from '../common/error';
+import { fetchMapAST, fetchProfileAST } from '../common/http';
+import { LogCallback } from '../common/log';
+import { ProfileId } from '../common/profile';
+import { ProfileMapReport } from '../common/report.interfaces';
+import { checkMapAndProfile, CheckResult } from './check';
+import { findLocalMapSource, findLocalProfileSource } from './check.utils';
+import { createProfileMapReport } from './lint';
+
+export function prePublishCheck(
+  profileAst: ProfileDocumentNode,
+  mapAst: MapDocumentNode,
+  providerJson: ProviderJson,
+  options?: { logCb?: LogCallback; warnCb?: LogCallback }
+): CheckResult[] {
+  try {
+    options?.logCb?.('Asserting profile document');
+    assertProfileDocumentNode(profileAst);
+  } catch (error) {
+    throw userError(error, 1);
+  }
+  try {
+    options?.logCb?.('Asserting map document');
+    assertMapDocumentNode(mapAst);
+  } catch (error) {
+    throw userError(error, 1);
+  }
+
+  //Check map and profile
+  const result = checkMapAndProfile(profileAst, mapAst, options);
+
+  try {
+    parseProviderJson(providerJson);
+  } catch (error) {
+    //TODO: better way of formating?
```

Comment:
this is super ugly, could you rather write a type guard?

Reply:
Fixed it with checking if error is ZodError (I needed zod which is in sdk anyway so i added it to CLI).